198. House Robber.java

```diff
@@ -0,0 +1,30 @@
+public class Solution {
+    public int rob(int[] nums) {
+        int output = 0;
+        HashMap<Integer, Integer> hm = new HashMap<Integer, Integer>();
```

Comment:
Generally you want the type of your object to be of the interface, because its more generic.
So instead of `HashMap<K, V> hm = new HashMap<K, V>();`, `Map<K, V> hm = new HashMap<K, V>()`.
The exception is when you have specific functionality that is only available in the derived class.